src/index.js

```diff
@@ -18,11 +20,13 @@ ReactDOM.render(
         <Route path="about" component={About} />
         <Route path="timers" component={TimersPage} />
         <Route path="admin" component={AdminPage} />
-        <Route path="invoice" component={InvoicePage} />
+        <Route path="invoice" component={NewInvoicePage} />
         <Route path="invoices" component={InvoicesPage} />
+        <Route path="invoices/:invoiceId" component={InvoicePage} 
+          onEnter={(nextState) => {store.dispatch(getInvoice(nextState.params.invoiceId));}} />
```

Comment:
Because this feature doesn't work when we navigate to the URL directly anyway, I think this onEnter callback should go away. It is not adding any value.